pkg/accounting/live/live.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package live
+
+import (
+	"context"
+	"strings"
+	"sync"
+
+	"github.com/skyrings/skyring-common/tools/uuid"
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+)
+
+// Config contains configurable values for the live accounting service.
+type Config struct {
+	StorageBackend string `help:"what to use for storing real-time accounting data"`
+}
+
+// Service represents the external interface to the live accounting
+// functionality.
+type Service interface {
+	GetProjectStorageUsage(ctx context.Context, projectID uuid.UUID) (int64, int64, error)
+	AddProjectStorageUsage(ctx context.Context, projectID uuid.UUID, inlineSpaceUsed, remoteSpaceUsed int64) error
+	ResetTotals()
+}
+
+// New creates a new live.Service instance of the type specified in
+// the provided config.
+func New(log *zap.Logger, config Config) (Service, error) {
```

Comment:
For data backend I've been trying to pass it as an argument to peer rather than create a direct dependency on peers. This ensure we don't create a hard dependency. Also it makes more obvious when we are touching disk.

Reply:
Sorry- pass what as an argument to peer? What peer?

Reply:
I mean `satellite.Peer`.

Currently this seems to implement persistance layer, which is in the realm of `satellitedb`, so it seems like it might fit more there. But, I agree it doesn't fit that mold exactly either since it's something inbetween a service and a cache.

Reply:
I think I've used the same patterns as all the other service-type things in peer.go ( https://github.com/storj/storj/blob/f74001ae9066fc84239f9cc563625793a24458df/satellite/peer.go#L311-L319 ). Is that what you mean?

Reply:
I was mainly wondering whether this actually should be part of the database.